imports/api/employee.js

```diff
@@ -0,0 +1,234 @@
+	if (!employee) {
+		throw new Meteor.Error(EmployeeError.EMPLOYEE_NOT_EXIST);
+	}
+	if (removedId !== this.employeeId) {
+		throw new Meteor.Error(AuthError.NO_PRIVILEGE);
+	}
+
+	return db.remove(
+		{
+			employeeId,
+		},
+		err => {
+			if (err) {
+				throw new Meteor.Error(EmployeeError.EMPLOYEE_REMOVE_FAIL);
+			} else {
+				let { tasksAssienedId } = employee;
+				tasksAssienedId = [...tasksAssienedId];
+				removeRelatedTasks(tasksAssienedId, employeeId, Tasks);
+				
+				let { tasksCreatedId } = employee;
+				tasksCreatedId = [...tasksCreatedId];
+				removeRelatedTasks(tasksCreatedId, employeeId, Tasks);
+
+				let { tasksWatchingId } = employee;
+				tasksWatchingId = [...tasksWatchingId];
+				removeRelatedTasks(tasksWatchingId, employeeId, Tasks);
+			}
+		},
+	);
+};
+
+export const createTask = (db, _id, taskId) => {
+	if (!this.userId) {
+		throw new Meteor.Error(AuthError.NOT_AUTH);
+	}
+	// TODO: not too sure use _id or employeeid
+	const employee = db.findOne({ _id });
+	if (!employee) {
+		throw new Meteor.Error(EmployeeError.EMPLOYEE_NOT_EXIST);
+	}
+	let { tasksCreatedId } = employee;
+	tasksCreatedId = [...tasksCreatedId];
+	if (!tasksCreatedId.includes(taskId)) {
+		tasksCreatedId = addToList(tasksCreatedId, taskId);
+		return db.update(
+			{
+				_id,
+			},
+			{
+				tasksCreatedId,
+			},
+			err => {
+				if (err) {
+					throw new Meteor.Error(EmployeeError.TASK_NOT_CREATABLE);
+				}
+			},
+		);
+	}
+
+	return db.update(_id, { tasksCreatedId });
+};
+
+export const removeCreateTaskFromEmployee = (db, _id, taskId) => {
+	if (!isAuthenticated()) {
+		throw new Meteor.Error(AuthError.NOT_AUTH);
+	}
+
+	const employee = db.findOne({ _id });
+	if (!employee) {
+		throw new Meteor.Error(EmployeeError.EMPLOYEE_NOT_EXIST);
+	}
+	let { tasksCreatedId } = employee;
+	tasksCreatedId = [...tasksCreatedId];
+	if (tasksCreatedId.includes(taskId)) {
+		tasksCreatedId = removeElement(tasksCreatedId, taskId);
+		return db.update(
+			{
+				_id,
+			},
+			{
+				tasksCreatedId,
+			},
+			err => {
+				if (err) {
+					throw new Meteor.Error(
+						EmployeeError.CREATED_TASK_NOT_REMOVABLE,
+					);
+				}
+			},
+		);
+	}
+
+	// TODO:
+};
+
+export const removeAssignedTaskFromExployee = () => {
+	if (!isAuthenticated()) {
+		throw new Meteor.Error(AuthError.NOT_AUTH);
+	}
+
+	// TODO:
+};
+
+export const removeWatchedTaskFromEmployee = () => {
+	if (!isAuthenticated()) {
+		throw new Meteor.Error(AuthError.NOT_AUTH);
+	}
+
+	// TODO:
+};
+
+export const assignTaskToEmployee = () => {
+	if (!isAuthenticated()) {
+		throw new Meteor.Error(AuthError.NOT_AUTH);
+	}
+
+	// TODO:
+};
+
+export const watchTask = () => {
+	if (!isAuthenticated()) {
+		throw new Meteor.Error(AuthError.NOT_AUTH);
+	}
+
+	// TODO:
+};
+
+export const unwatchTask = () => {
+	if (!isAuthenticated()) {
+		throw new Meteor.Error(AuthError.NOT_AUTH);
+	}
+
+	// TODO:
+};
+
+Meteor.methods({
+	[EMPLOYEESAPI.INSERT](_id, firstName, lastName, role) {
```

Comment:
For all the _id in the Meteor methods, let's have a getId helper in the authUtil. Since we use Meteor account-base for accountId, the id we initially pass in will be from Meteor.userId(), which corresponds to accountId in the employeesDB. So let's have something like getId in the authUtil like
`
const getId = (accountId) => {
    const employee = db.findOne({ accountId });
    return employee._id;
}
`
So when we call methods like insert, we can to Meteor.call(EMPLOYEESAPI.INSERT, "firstname", "lastname")
Role will always be ROLE.EMPLOYEE, and _id can be retrieved be the getId method in the definition of the method.
something like 
`
[EMPLOYEESAPI.INSERT](firstName, lastName) {
    const _id = getId(this.userId);
    return insertEmployee(Employees, _id, firstName, lastName, ROLE.EMPLOYEE);
}
`